src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,179 @@
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+
+        When job is running, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        5987779 regular_m submit_t      tgu  R       1:20      2 nid[005689-005690]
+
+        When job is finished/failed:
+            * job status is cached for a period of time (about 10 mins). squeue command will return code 0 and
+            out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+            * after the period of time, squeue command will return error code 1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+        """
+
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
+        if os.path.isfile(done_file):
+            return JobStatus.COMPLETED
+
+        # check job status using squeue command
+        std_out_file, std_err_file, exit_code = tf_common.run_nersc_command(
+            ['squeue', '-j', str(job_id)], self.job_dir, log_file_prefix='squeue', check_return_code=False)
+
+        if exit_code != 0:
+            # job finished without creating the done file indicating the job is failed
+            return JobStatus.FAILED
+
+        with open(std_out_file, "r") as std_out, open(std_err_file, "r") as std_err:
+            squeue_out = std_out.read().strip()
+            if str(job_id) in squeue_out:
+                return JobStatus.RUNNING
+            else:
+                return JobStatus.PENDING
+
+    def __init__(self, kbase_collection, load_ver, tool, root_dir=loader_common_names.ROOT_DIR,
+                 destroy_old_job_dir=False):
+        """
+        Initialize the task manager.
+
+        :param kbase_collection: KBase collection identifier name (e.g. GTDB).
+        :param load_ver: collection load version (e.g. r207.kbase.1).
+        :param tool: tool name (e.g. gtdb_tk, checkm2, etc.)
+        :param root_dir: root directory of the collection project.
+                         Default is the ROOT_DIR defined in src/loaders/common/loader_common_names.py
+        :param destroy_old_job_dir: if True, remove contents of the old job directory.
+        """
+
+        self.kbase_collection = kbase_collection
+        self.load_ver = load_ver
+        self.tool = tool
+        self.root_dir = root_dir
+
+        # job directory is named as <kbase_collection>_<load_ver>_<tool>
+        self.job_dir = os.path.join(self.root_dir, tf_common.TASKFARMER_JOB_DIR,
+                                    f'{self.kbase_collection}_{self.load_ver}_{self.tool}')
+        self.task_exists = os.path.isdir(self.job_dir) and os.listdir(self.job_dir)
+        self._create_job_dir(destroy_old_job_dir=destroy_old_job_dir)
+        self.tasks_df = self._retrieve_all_tasks()
```

Comment:
What's confusing to me here is that a client can call the `append_task_info` method, essentially creating a running task, IIUC, and get `task_exists` will remain false. I feel like I'm missing something here.

Reply:
👍  you are right. added updating task status after creating the task record.

Reply:
:+1: